packages/api/src/objectSet/ObjectSet.test.ts

```diff
@@ -52,13 +80,127 @@ describe("ObjectSet", () => {
     });
   });
 
+  describe("includeAllBaseObjectProperties", () => {
+    it("has the right types if you pass true", async () => {
+      const fetchPageResult = await fauxObjectSet
+        .where({ class: "idk" })
+        .fetchPage({ $includeAllBaseObjectProperties: true });
+
+      expectTypeOf(fetchPageResult).toEqualTypeOf<
+        PageResult<
+          Osdk.Instance<EmployeeApiTest, "$allBaseProperties">
+        >
+      >();
+
+      const asyncIterResult = fauxObjectSet
+        .where({ class: "idk" })
+        .asyncIter({ $includeAllBaseObjectProperties: true });
+
+      expectTypeOf(asyncIterResult).toEqualTypeOf<
+        AsyncIterableIterator<
+          Osdk.Instance<EmployeeApiTest, "$allBaseProperties">
+        >
+      >();
+    });
+
+    it("does not let you pass partial $select and true", async () => {
+      const fetchPageResult = await fauxObjectSet
+        .where({ class: "idk" })
+        .fetchPage({
+          // @ts-expect-error
+          $includeAllBaseObjectProperties: true,
+          $select: ["attachment"],
+        });
+
+      const asyncIterResult = fauxObjectSet
+        .where({ class: "idk" })
+        .asyncIter({
+          // @ts-expect-error
+          $includeAllBaseObjectProperties: true,
+          $select: ["attachment"],
+        });
+    });
+
+    it("does let you pass full select options and false", async () => {
+      const fetchPageResult = await fauxObjectSet
+        .where({ class: "idk" })
+        .fetchPage({
+          $includeAllBaseObjectProperties: true,
+
+          // this select list is intended to represent all properties on `EmployeeApiTest`,
+          // so if you get an error here later and you added properties to that object,
+          // be sure to add them here too.
+          $select: [
+            "attachment",
+            "class",
+            "employeeId",
+            "fullName",
+            "geopoint",
+            "geotimeSeriesReference",
+            "isActive",
+            "mediaReference",
+            "timeseries",
+          ],
+        });
+
+      const asyncIterResult = await fauxObjectSet
+        .where({ class: "idk" })
+        .fetchPage({
+          $includeAllBaseObjectProperties: true,
+
+          // this select list is intended to represent all properties on `EmployeeApiTest`,
+          // so if you get an error here later and you added properties to that object,
+          // be sure to add them here too.
+          $select: [
+            "attachment",
+            "class",
+            "employeeId",
+            "fullName",
+            "geopoint",
+            "geotimeSeriesReference",
+            "isActive",
+            "mediaReference",
+            "timeseries",
+          ],
```

Comment:
I have a version of this that just says you can have `$select` or $includeAllBaseObjectProperties` which we should consider.